backend/bitboxbase/handlers/handlers.go

```diff
@@ -74,38 +108,44 @@ func (handlers *Handlers) Uninit() {
 }
 
 func bbBaseError(err error, log *logrus.Entry) map[string]interface{} {
-	log.WithField("bbBaseError", err.Error()).Warning("Received an error from BitBox Base")
+	log.WithError(err).Error("Received an error from BitBox Base.")
+	if bbBaseError, ok := errp.Cause(err).(*rpcmessages.ErrorResponse); ok {
+		return map[string]interface{}{
+			"success": false,
+			"code":    bbBaseError.Code,
+			"message": bbBaseError.Message,
+		}
+	}
 	return map[string]interface{}{
 		"success": false,
```

Comment:
should this have an 'Unexpected' / 'Unknown' code?

Reply:
Yep, sure :)